File: packages/urdu-js/src/numbers/currency.ts
import type { Currency } from '../types.js'
import { numberToWords } from './number-to-words.js'

export function formatCurrency(amount: number, currency: Currency): string {
  const neg = amount < 0
  const abs = Math.abs(amount)
  const intPart = Math.trunc(abs)
  const fracPart = Math.round((abs - intPart) * 100)
  const intWords = numberToWords(BigInt(intPart))
  const currWord = currency === 'PKR' ? 'روپے' : 'روپیہ'
  const paiseWord = currency === 'PKR' ? 'پیسے' : 'پیسہ'
  let result = `${intWords} ${currWord}`
  if (fracPart > 0) result += ` ${numberToWords(BigInt(fracPart))} ${paiseWord}`
  return neg ? `منفی ${result}` : result
}

File: packages/urdu-dart/lib/src/numbers/currency.dart
import '../types.dart';
import 'number_to_words.dart';

/// Formats a number as a currency string in Urdu words.
String formatCurrency(double amount, Currency currency) {
  final neg = amount < 0;
  final abs = amount.abs();
  final intPart = abs.truncate();
  final fracPart = ((abs - intPart) * 100).round();

  final intWords = numberToWords(intPart);
  final currWord = currency == Currency.pkr ? 'روپے' : 'روپیہ';
  final paiseWord = currency == Currency.pkr ? 'پیسے' : 'پیسہ';

  var result = '$intWords $currWord';
  if (fracPart > 0) {
    result += ' ${numberToWords(fracPart)} $paiseWord';
  }
  return neg ? 'منفی $result' : result;
}
